### PR TITLE
Use lolight over highlight.js

### PR DESCRIPTION
**Pull request by MilyMilo**
I've replaced [hightlight.js](https://github.com/highlightjs/highlight.js/) with [lolight](https://github.com/lrsjng/lolight)

This reduces the bundle size by around 1/3:

With `highlight.js`:
image

With `lolight`:
image

Whereas the syntax highlighting is very similar too:

With `highlight.js`, and github theme:
image

With `lolight`, and my custom css:
image

Another benefit of lolight is that it's a generic tokenizer, it will highlight (for better or for worse) code in every language, and with highlight we're operating with a small set of imported languages (for bundle size reasons).

The only drawback to lolight is that it's maintenance status is unknown, but overall it's just a single js file which we could always fork or even just copy.

### CODE REVIEW

**Comment by ColdHeat** [comment]
I feel like this is a good idea but I'm concerned about the lack of updates on lolight. I think maybe in the future we would just want to syntax highlight on the server side.

**Comment by ColdHeat** [comment]
Okay I am okay with this but I feel like we should provide a way to disable the built in highlighter so that people could add in highlight js on their own. Could you add a setting option inside of the theme settings that disables our built in syntax highlighter? How does that sound?

**Comment by MilyMilo** [comment]
Needs this to be merged first: https://github.com/CTFd/CTFd/pull/2172

File: templates/config.html [hunk 1]
```diff
@@ -31,5 +31,17 @@
     <textarea class="form-control text-monospace" name="challenge_order" rows="5"></textarea>
   </div>
 
-  <button type="submit" class="btn btn-primary float-right">Update</button>
+  <div>
+    <div class="form-check">
+      <input type="checkbox" class="form-check-input" id="use_default_code_highlighter" name="use_default_code_highlighter" checked>
+      <label class="form-check-label" for="use_default_code_highlighter">
+        Use Default Code Highlighter
```
Comment (ColdHeat) [inline templates/config.html:38]: Can we change this to "Use Built-In Syntax Highlighter"

File: templates/config.html [hunk 2]
```diff
@@ -31,5 +31,17 @@
...
+      </label>
+    </div>
+    <small class="form-text text-muted">
+      Enable or disable the default code highlighter <a href="https://github.com/lrsjng/lolight" target="_blank">lolight</a>. This can be disabled to use your own code highlighter of choice.
```
Comment (ColdHeat) [inline templates/config.html:42]: Can you change the text for this to say something like: 

"The built-in syntax highlighter is lightweight and fast but may not always correctly highlight all languages. You can disable the built-in highlighter in order to use your own."

**Review by ColdHeat — APPROVED** [review]
Im pretty sure this is fine but just address the comments I left here if you could

**Pull request merged**